.github/workflows/ci-cpp.yaml

```diff
@@ -87,7 +87,7 @@ jobs:
         if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
         run: |
           echo "Build C++ client library"
-          export CMAKE_ARGS="-DCMAKE_BUILD_TYPE=Debug -DBUILD_DYNAMIC_LIB=OFF -DPYTHON_INCLUDE_DIR=/usr/include/python2.7 -DPYTHON_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython2.7.so"
+          export CMAKE_ARGS="-DCMAKE_BUILD_TYPE=Debug -DBUILD_DYNAMIC_LIB=OFF -DPYTHON_INCLUDE_DIR=/usr/include/python3.5m -DPYTHON_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython3.5m.so"
```

Comment:
I am having trouble with this test. I get the following error when running in a docker container on my laptop:

```
root@a89a873f16b0:/pulsar/pulsar-client-cpp# cd /pulsar/pulsar-client-cpp && cmake . $CMAKE_ARGS && make check-format && make -j8
-- ARCHITECTURE: x86_64
-- BUILD_DYNAMIC_LIB:  OFF
-- BUILD_STATIC_LIB:  ON
-- BUILD_TESTS:  ON
-- BUILD_PYTHON_WRAPPER:  ON
-- BUILD_WIRESHARK:  OFF
-- BUILD_PERF_TOOLS:  OFF
-- LINK_STATIC:  OFF
-- USE_LOG4CXX:  OFF
-- CMAKE_BUILD_TYPE:  Debug
-- Threads library: -pthread
Failed to find Protobuf in config mode, try to find it from system path
-- Protobuf_LIBRARIES: /usr/local/lib/libprotobuf.so
-- Protobuf_INCLUDE_DIRS: /usr/local/include
-- Found Boost: /usr/include (found version "1.58.0")  
-- Linking with Boost:System
-- Using std::regex
-- Found Boost: /usr/include (found version "1.58.0") found components: system 
-- PYTHON: 3.5.2
-- DETECTED Python 3
CMake Warning at /cmake-3.22.0-linux-x86_64/share/cmake-3.22/Modules/FindBoost.cmake:2201 (message):
  No header defined for python3-mt; skipping header check (note: header-only
  libraries have no designated component)
Call Stack (most recent call first):
  CMakeLists.txt:280 (find_package)


CMake Warning at /cmake-3.22.0-linux-x86_64/share/cmake-3.22/Modules/FindBoost.cmake:2201 (message):
  No header defined for python-py35; skipping header check (note: header-only
  libraries have no designated component)
Call Stack (most recent call first):
  CMakeLists.txt:280 (find_package)


CMake Warning at /cmake-3.22.0-linux-x86_64/share/cmake-3.22/Modules/FindBoost.cmake:2201 (message):
  No header defined for python-py35; skipping header check (note: header-only
  libraries have no designated component)
Call Stack (most recent call first):
  CMakeLists.txt:291 (find_package)


-- Found Boost: /usr/include (found version "1.58.0") found components: python-py35 
-- HAS_ZSTD: 1
-- HAS_SNAPPY: 1
-- Using Boost Python libs: 
CMake Error at python/CMakeLists.txt:84 (MESSAGE):
  Could not find Boost Python library


-- Configuring incomplete, errors occurred!
See also "/pulsar/pulsar-client-cpp/CMakeFiles/CMakeOutput.log".
See also "/pulsar/pulsar-client-cpp/CMakeFiles/CMakeError.log".
```

It is likely that `apachepulsar/pulsar-build:ubuntu-16.04-pb3` does not have the correct dependencies.